src/content/posts/multiplayer_online_game.md

```diff
@@ -0,0 +1,539 @@
+author: Zyno
+draft: false
+---
+
+This project, called Haunted Chronicles, started when we wanted to introduce ourselves to online multiplayer games and the code behind it.
+
+Naturally, we decided to code using python because it was simpler to begin with - everyone knew how to code in Python - and because we just wanted to discover the notion, not to code an AAA game.
+
+So, we began with a little documentation and we discovered the magic of **sockets**!
+
+For those who don't know anything about them, it is basically a glass bottle in which you put your message, and that you then throw away in the approximate direction of your friend, hoping for them to receive it. (Here is the very looong documentation of python : https://docs.python.org/3/library/socket.html)
+
+## First Step : Successfully sending a simple message to another computer in our LAN
+The first step here to understand how all this socket-stuff works is to try to make a simple 'email' system. The goal here is to make a python script able to send a predefined message to another computer.
+
+To do so, we need to define a server script and a client script.
+The server will initialize a socket which will listen to incoming messages, while the client will initialize a socket to send messages to the server.
+We wrote this code thanks to the [python documentation's example](https://docs.python.org/3/library/socketserver.html#socketserver-tcpserver-example)
+
+### The server side will look like this :
+```py
+import socketserver
+
+class MyTCPHandler(socketserver.BaseRequestHandler):
+    """
+    The request handler class for our server.
+
+    It is instantiated once per connection to the server, and must
+    override the handle() method to implement communication to the
+    client.
+    """
+
+    def handle(self):
+        # self.request is the TCP socket connected to the client
+        self.data = self.request.recv(1024).strip()
+        
+        in_ip = self.client_address[0]
+        
+        print("{} wrote:".format(in_ip))
+        in_data = str(self.data,'utf-16')
+        print(in_data)
+        
+        out = "Hello client, you correctly sent your message : '" + in_data + "' to the server."
+
+        print(">>> ",out,"\n")
+        self.request.sendall(bytes(out,'utf-16'))
+
+
+
+# ----------------------- Main -----------------------
+
+if __name__ == "__main__":
+    HOST, PORT = str(IP), 9998
+    socketserver.TCPServer.allow_reuse_address = True
+    # Create the server, bound to the given IP on port 9998
+    with socketserver.TCPServer((HOST, PORT), MyTCPHandler) as server:
+        print("HOST = ",IP,"\nPORT = ",PORT,"\n")
+        # Activate the server; this will keep running until you
+        # interrupt the program with Ctrl-C
+        server.serve_forever()
+```
+
+Ok so it may seems a big difficult to understand, but not everything here is important to really understand, and you will see further that what we did was in fact easier to understand in the end.
```

Comment:
```suggestion
Ok so it may seem a big difficult to understand, but not everything here is important to really understand, and you will see further that what we did was in fact easier to understand in the end.
```
If I'm not mistaken we do not conjugate verbs after modal verbs https://conjugator.reverso.net/conjugation-modals-auxiliaries-verb-english.html
>  Modal verbs are a special type of auxiliary verbs. They are always followed by the infinitive of a main verb without “to”, e.g. “You can leave now” (to leave), “She might know about this already” (to know).